models/DataObject/ClassDefinition/Data/Relations/AbstractRelations.php

```diff
@@ -420,75 +420,44 @@ public function loadLazyBrickField(DataObject\Objectbrick\Data\AbstractData $ite
-                || $container instanceof DataObject\Objectbrick\Data\AbstractData)
-                && !$container->isFieldDirty('_self'))
-        ) {
-            return $data;
-        }
-
-        if (!method_exists($this, 'getAllowMultipleAssignments') || !$this->getAllowMultipleAssignments()) {
-            $relationItems = [];
-            $objectId = null;
-            $fieldName = $this->getName();
-
-            if ($container instanceof DataObject\Concrete) {
-                $objectId = $container->getId();
-            } elseif (
-                    $container instanceof DataObject\Fieldcollection\Data\AbstractData ||
-                    $container instanceof DataObject\Localizedfield ||
-                    $container instanceof DataObject\Objectbrick\Data\AbstractData
-                ) {
-                $objectFromContainer = $container->getObject();
-                if ($objectFromContainer) {
-                    $objectId = $objectFromContainer->getId();
-                }
-            }
-
-            foreach ($data as $item) {
-                $elementHash = null;
-                if ($item instanceof DataObject\Data\ObjectMetadata || $item instanceof DataObject\Data\ElementMetadata) {
-                    if ($item->getElement() instanceof Element\ElementInterface) {
-                        $elementHash = Element\Service::getElementHash($item->getElement());
+        if (is_array($data)) {
+            if (!method_exists($this, 'getAllowMultipleAssignments') || !$this->getAllowMultipleAssignments()) {
+                $relationItems = [];
+                $fieldName = $this->getName();
+
+                foreach ($data as $item) {
+                    $elementHash = null;
+                    if ($item instanceof DataObject\Data\ObjectMetadata || $item instanceof DataObject\Data\ElementMetadata) {
+                        if ($item->getElement() instanceof Element\ElementInterface) {
+                            $elementHash = Element\Service::getElementHash($item->getElement());
+                        }
+                    } elseif ($item instanceof Element\ElementInterface) {
+                        $elementHash = Element\Service::getElementHash($item);
                     }
-                } elseif ($item instanceof Element\ElementInterface) {
-                    $elementHash = Element\Service::getElementHash($item);
-                }
 
-                if ($elementHash === null) {
-                    $relationItems[] = $item; //do not filter if element hash fails
-                } elseif (!isset($relationItems[$elementHash])) {
-                    $relationItems[$elementHash] = $item;
-                } else {
-                    throw new \Exception(sprintf('Passing relations multiple times is not allowed: %s in field %s of object id: %s', $elementHash, $fieldName, $objectId));
+                    if ($elementHash === null) {
+                        $relationItems[] = $item;
```

Comment:
throw new \Exception('Passing relations without ID or type not allowed anymore!);

Reply:
👍  for this change, since it is not possible to save relations without ID. please see https://github.com/pimcore/pimcore/issues/5953